backend/Infraestructure/Repositories/GenericRepository.cs

```diff
@@ -17,13 +17,13 @@ public GenericRepository(ApplicationDbContext context)
         {
             _context = context;
         }
-        public async Task AddAsync(T entity)
+        public virtual async Task AddAsync(T entity)
```

Comment:
The GenericRepository shouldn't be modified, please don't add virtual, since it's not going to override every time is used, it's going to be inherit and you will use it directly, please remove virtual to every method, this is not polymorphism, is inheritance.